tiledb/sm/serialization/array_schema.cc

```diff
@@ -2081,26 +2082,22 @@ Status max_buffer_sizes_deserialize(
 
 void serialize_load_array_schema_request(
     const Config&, const LoadArraySchemaRequest&, SerializationType, Buffer&) {
-  throw Status_SerializationError(
-      "Cannot serialize; serialization not enabled.");
+  throw GenericException("Cannot serialize; serialization not enabled.");
```

Comment:
For this exception see, for example, `class ArraySerializationDisabledException`. We should use the analogous technique here.

I'm not sure we need an exception type for each kind of object being serialized. I'd be good with a single `SerializationDisabledException` with origin `serialization`.

Reply:
I did not want to deviate from the already used techniques so I created all the necessary ```*DisabledException``` classes

Reply:
>I created all the necessary *DisabledException classes

That's fine too.